internal/providers/system/system.go

```diff
@@ -30,12 +31,24 @@ import (
 )
 
 const (
-	baseFilename = "base.ign"
 	userFilename = "user.ign"
 )
 
-func FetchBaseConfig(logger *log.Logger) (types.Config, report.Report, error) {
-	return fetchConfig(logger, baseFilename)
+func FetchBaseConfig(logger *log.Logger, platformName string) (types.Config, report.Report, error) {
+	var fullBaseConfig types.Config
+	baseDConfig, fullReport, err := fetchBaseDirectoryConfig(logger, "base.d")
```

Comment:
`baseDConfig` is still unnecessary.  Per my prior comment, you can just assign directly to `fullBaseConfig`.